crates/uv/src/commands/tool/install.rs

```diff
@@ -171,9 +185,15 @@ pub(crate) async fn install(
     // entrypoints (without `--force`).
     let reinstall_entry_points = existing_tool_receipt.is_some();
 
-    // TODO(zanieb): Build the environment in the cache directory then copy into the tool directory
-    // This lets us confirm the environment is valid before removing an existing install
-    let environment = installed_tools.environment(&from.name, force, interpreter, cache)?;
+    // TODO(zanieb): Build the environment in the cache directory then copy into the tool directory.
+    // This lets us confirm the environment is valid before removing an existing install. However,
+    // entrypoints always contain an absolute path to the relevant Python interpreter, which would
+    // be invalidated by moving the environment.
```

Comment:
Oh.. hm.

Reply:
Yeah... we've talked before about writing relative paths in this specific case (with opt-in) to make scripts relocatable.

Reply:
We can also have a special entry point that calls into `uv` to determine what to do (I've kind of wanted this for other reasons)

Reply:
It's not very hard but I don't know what assumptions it would break downstream.

Reply:
Seems really annoying to nuke the environment before we solve the requirements. I guess we should create a couple tracking issues for these things.

Reply:
We should probably solve before we do this.

Reply:
https://github.com/astral-sh/uv/issues/4747

Reply:
We actually already have an interpreter and everything so we could even solve before finding the environment etc.